part2/stages/Functions/install-main

```diff
@@ -315,6 +322,27 @@ install_dom0()
     finalize_keys "${RECOVERY_KEY}" "${PLATFORM_KEY}" "${CONFIG_KEY}" || return 1
 }
 
+# Minimal step to get the upgrae-compat rootfs unpacked
+# and installed as a LV, but only if we're upgrading from
+# a 32-bit machine.  We'll clean up the LV later
+install_upgradecompat()
+{
+
+    local UPGRADE_ROOTFS="$1"
+    local ROOTFS_TYPE="$2"
+    COMPAT_SIZE_BYTES=$(zcat ${UPGRADE_ROOTFS} | wc -c)
+    COMPAT_LV_SIZE=$(((${COMPAT_SIZE_BYTES} / 1048576) + 1))M
+
+    if [ "$(uname -m)" = "i686" ]; then
+		do_cmd create_lv "xenclient" "upgradecompat" \
+			--size ${COMPAT_LV_SIZE} >&2 || return 1
+		write_rootfs ${UPGRADE_ROOTFS} ${ROOTFS_TYPE} /dev/xenclient/upgradecompat >&2 || {
+			do_cmd lvremove -f /dev/xenclient/upgradecompat >&2
+			return 1
+		}
+    fi
```

Comment:
Mixed tabs/spaces indentation.
Also, to prevent failures with `UPGRADE_ROOTFS`, `ROOTFS_TYPE`, `COMPAT_LV_SIZE`, `COMPAT_SIZE_BYTES` being invalid, empty or including a space due to error output, failure, planet alignment, etc, the call sites and definitions should be quoted.